File: stack&queue/leetcode_20/solution.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        mapping = {
            "(":")",
            "{":"}",
            "[":"]"
        }

        for char in s:
            if mapping.get(char):
                stack.append(mapping[char])
            else:
                if not stack or stack.pop() != char:
                    return False

        return not stack

File: stack&queue/leetcode_20/solution.js
/**
 * @param {string} s
 * @return {boolean}
 */

var isValid = function(s) {
    const stack= [];
    const map = {
        "(":")",
        "[":"]",
        "{":"}"
    }

    for (let i=0; i<s.length; i++) {
        const char = s[i];
        
        if(map[char]){
            stack.push(map[char])
        }else{
            if(!stack.length || stack.pop() !== char){
                return false
            }
        }
    }
  
    return stack.length === 0
};